### PR TITLE
Support spend_unconfirmed in EstimateFee and FundPsbt

### PR DESCRIPTION
**Pull request by mrfelton**
This is a rebased version of https://github.com/lightningnetwork/lnd/pull/4845. I'm not sure what happened to that original PR but it seems the repo where it came from is no longer online.

`FundPsbt` and `EstimateFee` now support spending from the underlying wallet using unconfirmed inputs or inputs with a specific number of minumum confirmations; this is made possible by adding the params `min_confs` and `spend_unconfirmed` to the underlying proto messages, and implementing checks in their function calls.

It changes the interface of `WalletController` slightly; it adds the param `minConfs` to `CreateSimpleTx` inside the `WalletController` interface, thus allowing the underlying implementation of the function to pass `minConfs` to its underlying wallet's `createsimpletx` implementation (if possible and/or desired).

### CODE REVIEW

File: lnrpc/rpc.proto [hunk 1]
```diff
@@ -891,6 +891,13 @@ message EstimateFeeRequest {
     // The target number of blocks that this transaction should be confirmed
     // by.
     int32 target_conf = 2;
+
+    // The minimum number of confirmations each one of your outputs used for
+    // the transaction must satisfy.
+    int32 min_confs = 3;
```
Comment (guggero) [inline lnrpc/rpc.proto:897]: Please split the changes in the main RPC (`rpc.proto` and `rpcserver.go`) into a separate commit and the changes in the `walletrpc` package into another.

File: lnrpc/walletrpc/walletkit_server.go [hunk 2]
```diff
@@ -981,6 +981,24 @@ func (w *WalletKit) FundPsbt(_ context.Context,
 			"specify either target_conf or set_per_vbyte")
 	}
 
+	var (
+		// minConfs can be provided by the RPC request,
+		// but defaults to defaultMinConf (1).
+		minConfs = int32(defaultMinConf)
+	)
+
+	switch {
```
Comment (guggero) [inline lnrpc/walletrpc/walletkit_server.go:990]: Should use `lnrpc.ExtractMinConfs()` instead.

File: lnwallet/btcwallet/btcwallet.go [hunk 3]
```diff
@@ -333,7 +333,8 @@ func (b *BtcWallet) SendOutputs(outputs []*wire.TxOut,
 //
 // This is a part of the WalletController interface.
 func (b *BtcWallet) CreateSimpleTx(outputs []*wire.TxOut,
-	feeRate chainfee.SatPerKWeight, dryRun bool) (*txauthor.AuthoredTx, error) {
+	feeRate chainfee.SatPerKWeight, minConfs int32,
```
Comment (guggero) [inline lnwallet/btcwallet/btcwallet.go:336]: This change should be its own commit.

File: lnwallet/interface.go [hunk 4]
```diff
@@ -187,7 +187,7 @@ type WalletController interface {
 	//
 	// NOTE: This method requires the global coin selection lock to be held.
 	SendOutputs(outputs []*wire.TxOut,
-		feeRate chainfee.SatPerKWeight, minconf int32, label string) (*wire.MsgTx, error)
+		feeRate chainfee.SatPerKWeight, minConfs int32, label string) (*wire.MsgTx, error)
```
Comment (guggero) [inline lnwallet/interface.go:190]: The renaming should also be a separate commit.

File: rpcserver.go [hunk 5]
```diff
@@ -1159,12 +1159,28 @@ func (r *rpcServer) EstimateFee(ctx context.Context,
 		return nil, err
 	}
 
+	// minConfs can be provided by the RPC request,
+	// but defaults to defaultMinConf (1).
+	var minConfs int32 = 1
+
+	switch {
```
Comment (guggero) [inline rpcserver.go:1166]: Same here, use `ExtractMinConfs`.

**Review by guggero — COMMENTED** [review]
Thank you for picking this up!
Did a first pass, looks pretty good.
Main comments are about splitting the commit and reusing existing code.

**Comment by alexbosworth** [comment]
I'd love to see this change, it would help me doing CPFPs a lot

**Comment by wpaulino** [comment]
Ping @mrfelton. If you're still available to work on this, we can try to get it merged for v0.13.

**Comment by mrfelton** [comment]
Sure, I'll take a look. Missed the last batch of updates on this one somehow. Have been running with this patch for some months though.

**Review by guggero — APPROVED** [review]
Very useful feature, thank you. LGTM :100:

**Review by wpaulino — APPROVED** [review]
